src/axom/core/StackArray.hpp

```diff
@@ -108,6 +108,33 @@ AXOM_HOST_DEVICE bool operator!=(const StackArray<T, N>& lhs,
   return !(lhs == rhs);
 }
 
+/*!
+ * \brief Less than operator for StackArray
+ *
+ * \param [in] lhs left StackArray to compare
+ * \param [in] rhs right StackArray to compare
+ * \return true if \a lhs is lexicographically less than \a rhs, false otherwise
+ * \note It is only valid to call this function when values of type \a T are comparable, 
+ * e.g. when T has an operator<()
+ */
+template <typename T, int N>
+AXOM_HOST_DEVICE bool operator<(const StackArray<T, N>& lhs,
+                                const StackArray<T, N>& rhs)
+{
+  for(int i = 0; i < N; ++i)
+  {
+    if(lhs[i] < rhs[i])
+    {
+      return true;
+    }
+    else if(lhs[i] > rhs[i])
```

Comment:
I think it would be more efficient to change this to '>='. For example, if lhs == rsh is true on entry, then the entire array has to be traversed.

Reply:
At one point, this was a bottleneck in the Delaunay point insertion, so I played around with several iterations of this lexicographic ordering function. 

The implementation here appeared to be a good compromise between clarity and efficiency (the compiler can optimize this pretty well since `N` is known at compile time and typically pretty small).

Would it be ok to make this a separate issue and add benchmarks to compare several variants of a comparator for `StackArray::operator<()`?

Reply:
It's your call. I probably have less faith in compilers than you.

Reply:
i sent kenny a teams message about this and he ignored me =*( i'm sad.